settings.js

```diff
@@ -302,28 +302,35 @@ function findConflicts(schemas) {
 */
 var winprops = [];
 
-function winprop_match_p(meta_window, prop) {
-    let wm_class = meta_window.wm_class || "";
-    let title = meta_window.title;
-    if (prop.wm_class.constructor === RegExp) {
-        if (!wm_class.match(prop.wm_class))
-            return false;
-    } else if (prop.wm_class !== wm_class) {
-        return false;
-    }
-    if (prop.title) {
-        if (prop.title.constructor === RegExp) {
-            if (!title.match(prop.title))
+function _winprop_match_p_helper(user_attr_prop, _win_prop) {
+    // matches user-supplied conditions with null checking
```

Comment:
A more descriptive name allows the comment to explain _why_, not just _what_
```suggestion
function condition_matches_winprop(condition, _win_prop) {
    // A null condition matches any value
```
(also requires corresponding name changes below, both in this function and where it is called)